### DISCUSSION TITLE
Reloading assemblies.

### DISCUSSION THREAD

**Discussion by Ggjorven** [StudioCherno/Coral#9] [Q&A] [answered]
**How do I reload a assembly?**

I'm creating a simple game engine and want to use C# scripting and since I knew this project existed I decided to use Coral.

In my script class I load the dll using this line:
`m_Assembly = s_Context.LoadAssembly(pathStr);`

The 's_Context' is a global Coral::AssemblyLoadContext, so all script's can share it.
When I call the line again with the same path it gives me this error:

```
[Coral](Error): System.ArgumentException: An item with the same key has already been added. Key: 1723769324
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Coral.Managed.AssemblyLoader.LoadAssembly(Int32 InContextId, NativeString InAssemblyFilePath)
```

What is the proper way of reloading in Coral?

Thanks in advance!

**Comment by peter1745** [comment, marked as answer]
You first have to unload the AssemblyLoadContext by calling `HostInstance::UnloadAssemblyLoadContext(s_Context)`, and then you have to recreate the assembly load context and load all the assemblies again

**Reply by Ggjorven** [reply]
Thanks a lot, I've tested it and works perfectly! Just a note for future people maybe looking at this post: make sure you destroy the objects that you've created from the previous assembly to avoid any warnings.

**Reply by peter1745** [reply]
Yeah it's good to clean up the objects yourself, although Coral *will* clean them up automatically when the ALC is unloaded. But yeah you should be cleaning them up yourself to be safe 🙂
